api/common/sharing/utils.go

```diff
@@ -106,21 +101,23 @@ func validateParameters(parameters map[string]gjson.Result) error {
 	if len(parameters) == 0 {
 		return util.HandleInstanceSharingError(util.ErrMissingOrInvalidReferenceParameter, "")
 	}
+	hasAtLeastOneSelector := false
 	ID, IDExists := parameters[instance_sharing.ReferencedInstanceIDKey]
-	name, nameExists := parameters[instance_sharing.ReferenceInstanceNameSelectorKey]
-	plan, planExists := parameters[instance_sharing.ReferencePlanNameSelectorKey]
-	labels, labelsExists := parameters[instance_sharing.ReferenceLabelSelectorKey]
-	selectorLabels := types.Labels{}
-	if labelsExists {
-		if err := util.BytesToObject([]byte(labels.Raw), &selectorLabels); err != nil {
-			return err
-		}
+	validID := IDExists && ID.Type == gjson.String && len(ID.String()) > 0
+
+	selectors, selectorsExists := parameters[instance_sharing.SelectorsKey]
+	if selectorsExists {
+		selectorsMap := selectors.Map()
+		name, nameExists := selectorsMap[instance_sharing.ReferenceInstanceNameSelectorKey]
+		plan, planExists := selectorsMap[instance_sharing.ReferencePlanNameSelectorKey]
+		labels, labelsExists := selectorsMap[instance_sharing.ReferenceLabelSelectorKey]
+		validName := nameExists && name.Type == gjson.String && len(name.String()) > 0
+		validPlan := planExists && plan.Type == gjson.String && len(plan.String()) > 0
+		validLabels := labelsExists && labels.String() != "" && len(labels.Array()) > 0
+		hasAtLeastOneSelector = validLabels || validName || validPlan
 	}
-	hasAtLeastOneSelector := selectorLabels != nil && len(selectorLabels) > 0 || nameExists && len(name.String()) > 0 || planExists && len(plan.String()) > 0
-	selectorsAndID := IDExists && len(ID.String()) > 0 && hasAtLeastOneSelector
-	emptyValues := len(ID.String()) == 0 && !hasAtLeastOneSelector
-	if selectorsAndID || emptyValues {
-		return util.HandleInstanceSharingError(util.ErrInvalidReferenceSelectors, "")
+	if (validID && hasAtLeastOneSelector) || (!validID && !hasAtLeastOneSelector) {
```

Comment:
if selectors doesn't exist you pass null  in validateParameters function, then  in line 40  you do the following:
selectors := parameters[instance_sharing.SelectorsKey].Map() 
in this case it will crash

Reply:
if `selectorsExists=false`, 
then `hasAtLeastOneSelector=false`
so either you are using the `referenced_instance_id` key and won't enter this selectors flow.
or you will fail on the condition of line 119: `validID && hasAtLeastOneSelector) || (!validID && !hasAtLeastOneSelector`

I cover the scenario you mention with tests:
Context: `empty selectors object`

Reply:
ok